plugins/serializers/splunkmetric/splunkmetric.go

```diff
@@ -0,0 +1,135 @@
+package splunkmetric
+
+import (
+	"encoding/json"
+	"errors"
+	"time"
+
+	"github.com/influxdata/telegraf"
+)
+
+type serializer struct {
+	TimestampUnits time.Duration
+}
+
+func NewSerializer(timestampUnits time.Duration) (*serializer, error) {
+	s := &serializer{
+		TimestampUnits: truncateDuration(timestampUnits),
+	}
+	return s, nil
+}
+
+func (s *serializer) Serialize(metric telegraf.Metric) ([]byte, error) {
+	var serialized string
+
+	m, err := s.createObject(metric)
+	if err == nil {
+		serialized = m + "\n"
+	}
+
+	return []byte(serialized), nil
+}
+
+func (s *serializer) SerializeBatch(metrics []telegraf.Metric) ([]byte, error) {
+
+	var serialized string
+
+	var objects []string
+
+	for _, metric := range metrics {
+		m, err := s.createObject(metric)
+		if err == nil {
+			objects = append(objects, m)
+		}
+	}
+
+	for _, m := range objects {
+		serialized = serialized + m + "\n"
+	}
+
+	return []byte(serialized), nil
+}
+
+func (s *serializer) createObject(metric telegraf.Metric) (metricString string, err error) {
+
+	/* Splunk supports one metric per line and has the following required names:
+	 ** metric_name: The name of the metric
+	 ** _value:      The value for the metric
+	 ** _time:       The timestamp for the metric
+	 ** All other index fields become deminsions.
```

Comment:
I personally don't think this is a good structure to follow. This will prevent using this serializer with pretty much every single input plugin, as none of them follow this format. I think the implementation over on #4185 has a much more flexible implementation.

Reply:
The comment is wrong, it should be `time`, not `_time` (I'll fix), but `metric_name` and `_value` are required fields for the metrics store. Naming the fields this way means you don't need custom props.conf or transforms.conf. In fact this is the same naming convention that is used in the PR you reference. The point of this serializer is to format the metrics into this format so that it can be used with the generic, well tested, HTTP output. Or the file output if you're running telegraf on a machine that is running a Splunk forwarder.